Documentation/Evolution/DelimiterSyntax.md

```diff
@@ -0,0 +1,388 @@
+
+A regex literal also allows for seamless composition with the Regex DSL, enabling lightweight intermixing of a regex syntax with other elements of the builder:
+
+```swift
+// A regex for extracting a currency (dollars or pounds) and amount from input 
+// with precisely the form /[$£]\d+\.\d{2}/
+let regex = Regex {
+  Capture { /[$£]/ }
+  TryCapture {
+    /\d+/
+    "."
+    /\d{2}/
+  } transform: {
+    Amount(twoDecimalPlaces: $0)
+  }
+}
+```
+
+This flexibility allows for terse matching syntax to be used when it's suitable, and more explicit syntax where clarity and strong types are required.
+
+Due to the existing use of `/` in comment syntax and operators, there are some syntactic ambiguities to consider. While there are quite a few cases to consider, we do not feel that the impact of any individual case is sufficient to disqualify the syntax. Some of these ambiguities require a couple of source breaking language changes, and as such the `/.../` syntax requires upgrading to a new language mode in order to use.
+
+## Detailed design
+
+### Upgrade path
+
+Due to the source breaking changes needed for the `/.../` syntax, it will be introduced in Swift 6 mode. However, projects will be able to adopt it earlier by using the compiler flag `-enable-regex-literals`. Note this does not affect the extended literal `#/.../#`, which will be usable immediately.
+
+### Named typed captures
+
+Regex literals have their capture types statically determined by the capture groups present. This follows the same inference behavior as [the DSL][regex-dsl], and is explored in more detail in *[Strongly Typed Captures][strongly-typed-captures]*. One aspect of this that is currently unique to the literal is the ability to infer labeled tuple elements for named capture groups. For example:
+
+```swift
+func matchHexAssignment(_ input: String) -> (String, Int)? {
+  let regex = /(?<identifier>[[:alpha:]]\w*) = (?<hex>[0-9A-F]+)/
+  // regex: Regex<(Substring, identifier: Substring, hex: Substring)>
+  
+  guard let match = regex.matchWhole(input), 
+        let hex = Int(match.hex, radix: 16) 
+  else { return nil }
+  
+  return (match.identifier, hex)
+}
+```
+
+This allows the captures to be referenced as `match.identifier` and `match.hex`, in addition to numerically (like unnamed capture groups) as `match.1` and `match.2`. This label inference behavior is not available in the DSL, however users are able to [bind captures to named variables instead][dsl-captures].
+
+### Extended delimiters `#/.../#`, `##/.../##`
+
+Backslashes may be used to write forward slashes within the regex literal, e.g `/foo\/bar/`. However, this can be quite syntactically noisy and confusing. To avoid this, a regex literal may be surrounded by an arbitrary number of balanced octothorpes. This changes the delimiter of the literal, and therefore allows the use of forward slashes without escaping. For example:
```

Comment:
We call `#` a pound symbol elsewhere in the proposal, should we be consistent?

Reply:
FWIW Unicode calls it "number sign", which is less confusable with £ (whose official name is "POUND SIGN"). How does "number sign" consistently sound? Even in the US, we don't dial numbers that often anymore, so I expect "pound sign" to have limited life span, especially in the era of hashtags.

Reply:
Updated to use "number sign", though a couple of occurrences were phrased as "number of ...", which sounded a bit awkward, so I changed those to use the `#` character directly.